tests/foreman/ui/test_repository.py

```diff
@@ -133,44 +148,46 @@ def test_create_repo_with_checksum(self, repo_name):
         # Creates new product
         product = entities.Product(organization=self.organization).create()
         with Session(self.browser) as session:
-            make_repository(
-                session,
-                org=self.organization.name,
-                name=repo_name,
-                product=product.name,
-                url=FAKE_1_YUM_REPO,
-                repo_checksum=checksum,
-            )
-            self.assertTrue(self.repository.validate_field(
-                repo_name, 'checksum', checksum))
+            for repo_name in generate_strings_list():
+                with self.subTest(repo_name):
+                    make_repository(
+                        session,
+                        org=self.organization.name,
+                        name=repo_name,
+                        product=product.name,
+                        url=FAKE_1_YUM_REPO,
+                        repo_checksum=checksum,
+                    )
+                    self.assertTrue(self.repository.validate_field(
+                        repo_name, 'checksum', checksum))
 
     @run_only_on('sat')
-    @data('', '   ')
-    def test_negative_create_with_blank(self, repo_name):
-        """@Test: Create repository with blank and whitespace in name
+    def test_negative_create_with_blank(self):
+        """@Test: Create repository with invalid name
 
-        @Feature: Content Repos - Negative Create zero length
+        @Feature: Content Repos - Negative Create
 
         @Assert: Repos is not created
 
         """
         # Creates new product
         product = entities.Product(organization=self.organization).create()
-        with Session(self.browser) as session:
-            make_repository(
-                session,
-                org=self.organization.name,
-                name=repo_name,
-                product=product.name,
-                url=FAKE_1_YUM_REPO,
-            )
-            invalid = self.products.wait_until_element(
-                common_locators['common_invalid'])
-            self.assertIsNotNone(invalid)
+        for repo_name in invalid_values_list(interface='ui'):
+            with self.subTest(repo_name):
+                with Session(self.browser) as session:
+                    make_repository(
+                        session,
+                        org=self.organization.name,
+                        name=repo_name,
+                        product=product.name,
+                        url=FAKE_1_YUM_REPO,
+                    )
+                    invalid = self.products.wait_until_element(
+                        common_locators['common_invalid'])
+                    self.assertIsNotNone(invalid)
 
     @run_only_on('sat')
-    @data(*generate_strings_list())
-    def test_negative_create_with_same_names(self, repo_name):
```

Comment:
I intentionally changed this test from data driven to normal test to avoid test redundancy